crates/lsp/src/types.rs

```diff
@@ -0,0 +1,187 @@
+    pub(crate) fetch: Option<bool>,
+
+    /// Tests runs can be requested from the server.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) run: Option<bool>,
+
+    /// The server will send notifications to update tests.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) update: Option<bool>,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct NargoCapability {
+    /// The server will provide various features related to testing within Nargo.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) tests: Option<NargoTestsOptions>,
+}
+
+#[derive(Debug, PartialEq, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct ServerCapabilities {
+    /// Defines how text documents are synced.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) text_document_sync: Option<TextDocumentSyncCapability>,
+
+    /// The server provides code lens.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) code_lens_provider: Option<CodeLensOptions>,
+
+    /// The server handles and provides custom nargo messages.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) nargo: Option<NargoCapability>,
+}
+
+#[derive(Debug, PartialEq, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct InitializeResult {
+    /// The capabilities the language server provides.
+    pub(crate) capabilities: ServerCapabilities,
+
+    /// Information about the server.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) server_info: Option<ServerInfo>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "String")]
+pub(crate) struct NargoTestId(CrateName, String);
+
+impl TryFrom<String> for NargoTestId {
+    type Error = String;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        if let Some((crate_name, func_name)) = value.split_once('/') {
+            let crate_name = crate_name.parse()?;
+            Ok(Self(crate_name, func_name.to_string()))
+        } else {
+            Err("NargoTestId should be serialized as package_name/fully_qualified_path".to_string())
+        }
+    }
+}
+
+impl From<NargoTestId> for String {
+    fn from(value: NargoTestId) -> Self {
+        format!("{}/{}", value.0, value.1)
+    }
+}
+
+impl NargoTestId {
+    pub(crate) fn new(crate_name: CrateName, func_name: String) -> Self {
+        Self(crate_name, func_name)
+    }
+
+    pub(crate) fn crate_name(&self) -> &CrateName {
+        &self.0
+    }
+
+    pub(crate) fn function_name(&self) -> &String {
+        &self.1
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct NargoTest {
+    pub(crate) id: NargoTestId,
+    /// Fully-qualified path to the test within the crate
+    pub(crate) label: String,
```

Comment:
If this is a fully qualified path which presumably can be used as a label, why not name it `fully_qualified_path_to_test` or something along those lines?